File: examples/python/pagination/_async/sandbox.py
import asyncio

from northrays import AsyncNorthrays, ListSandboxesQuery, SandboxListSortDirection, SandboxListSortField, SandboxState


async def main():
    async with AsyncNorthrays() as northrays:
        async for sandbox in northrays.list(
            ListSandboxesQuery(
                limit=10,
                labels={"env": "dev"},
                states=[SandboxState.STARTED],
                sort=SandboxListSortField.CREATEDAT,
                order=SandboxListSortDirection.DESC,
            )
        ):
            print(sandbox.id)


if __name__ == "__main__":
    asyncio.run(main())

File: examples/typescript/pagination/sandbox.ts
import { Northrays, SandboxListSortDirection, SandboxListSortField, SandboxState } from '@northrays/sdk'

async function main() {
  const northrays = new Northrays()

  for await (const sandbox of northrays.list({
    limit: 10,
    labels: { env: 'dev' },
    states: [SandboxState.STARTED],
    sort: SandboxListSortField.CREATED_AT,
    order: SandboxListSortDirection.DESC,
  })) {
    console.log(sandbox.id)
  }
}

main().catch(console.error)
